### ISSUE TITLE
Fix "Development Workflow" Wiki

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-che/che#15521]
https://github.com/eclipse/che/wiki/Development-Workflow,-Che-7 seems to be severely out of date. Someone needs to go over it an fix it/assign tasks for knowledgeable people to fix it.

**Comment by benoitf** [comment]
Assigning platform as document is about compiling che / producing the docker image

**Comment by sunix** [comment]
@benoitf @skabashnyuk @tsmaeder Should we simply remove that page ?
Contribution guide should go to CONTRIBUTING.md file of each repo.

**Comment by tsmaeder** [comment]
I don't think so: there needs to be a starting point that tells people about the global process of building Che. We use the che repository as a single point of contact also for issues, so I think this is where people will come to.

**Comment by sunix** [comment]
@skabashnyuk could someone from your team update https://github.com/eclipse/che/blob/HEAD/CONTRIBUTING.md#contributing-to-che ?
For building the che-server, it should probably goes in a dedicated CONTRIBUTING.md.

And we should redirecting people reading this wiki page to a more generic contribution page which does not include only che-server: https://github.com/eclipse/che/blob/HEAD/CONTRIBUTING.md
